openpathsampling/experimental/simstore/regressions/test_numpy_sfrs.py

```diff
@@ -0,0 +1,93 @@
+# This test checks a regression related to  storing results from storable
+# functions that are of not basic types that SQLAlchemy knows. In
+# particular, functions that returns NumPy arrays were storing the results
+# without complaint, but could not re-load them as the correct type.
+import pytest
+import numpy as np
+
+from openpathsampling.experimental.simstore import (
+    StorableFunction, StorableFunctionResults, storable_function_find_uuids
+)
+from openpathsampling.experimental.simstore import SQLStorageBackend
+from openpathsampling.experimental.simstore import GeneralStorage
+from openpathsampling.experimental.simstore.class_info import (
+    SerializationSchema, ClassInfo
+)
+from openpathsampling.netcdfplus import StorableObject
+from openpathsampling.experimental.storage.ops_storage import (
+    unsafe_ops_codecs, safe_ops_codecs
+)
+from openpathsampling.experimental.simstore.serialization_helpers import (
+    default_find_uuids
+)
+
+from openpathsampling.experimental.simstore.test_utils import MockUUIDObject
+
+
+@pytest.fixture
+def minimal_sfr_serialization():
+    schema = {
+        'foo': [('normal_attr', 'int')],
+        'storable_functions': [('json', 'json_obj')],
+        'simulation_objects': [('json', 'json_obj')],
+    }
+    serialization = SerializationSchema(
+        default_info=ClassInfo(
+            table='simulation_objects',
+            cls=StorableObject,
+            serializer=unsafe_ops_codecs.simobj_serializer,
+            deserializer=unsafe_ops_codecs.simobj_deserializer,
+            safe_deserializer=safe_ops_codecs.simobj_serializer,
+            find_uuids=default_find_uuids
+        ),
+        sfr_info=ClassInfo(
+            table="function_results",
+            cls=StorableFunctionResults,
+            serializer=StorableFunctionResults.to_dict,
+            deserializer=lambda x: x,  # deserializer not used
+            find_uuids=default_find_uuids
+        ),
+        schema=schema,
+        class_info_list=[
+            ClassInfo(table='foo', cls=MockUUIDObject),
+            ClassInfo(table='storable_functions',
+                      cls=StorableFunction,
+                      find_uuids=storable_function_find_uuids,
+                      serializer=unsafe_ops_codecs.simobj_serializer,
+                      deserializer=unsafe_ops_codecs.simobj_deserializer),
+        ]
+    )
+    return schema, serialization
+
+def stupid_func(inp):
+    import numpy as np
+    return np.array([1.0, 2.0, 3.0])
+
+def test_numpy_sfrs(minimal_sfr_serialization, tmpdir):
```

Comment:
I think both statements are true: `tmpdir` should work in 2.7, but we don't care because `experimental` tests don't run in 2.7.

According to the pytest docs, [tmpdir is a py.path](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture) object. `py.path` was `pathlib` before `pathlib`. I assume this was added to pytest while 2.7 was still supported.

However, the `experimental` subpackage makes no promise of 2.7 support (since this is stuff that is being offered as a preview before moving to the main code in OPS 2.0). CI specifically excludes 2.7 from experimental tests: https://github.com/openpathsampling/openpathsampling/blob/e8b091c92916561954542d40d17d7241b203d1ad/.github/workflows/tests.yml#L76-L78

Reply:
perfect, thanks for the clarification :+1: